src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -993,33 +993,23 @@ def getClassAds(self):
             logging.debug(msg)
             return None, None
         else:
-            for slicedAds in grouper(itobj, 1000):
-                for jobAd in slicedAds:
-                    ### This condition ignores jobs that are Removed, but stay in the X state
-                    ### For manual condor_rm removal, job wont be in the queue \
-                    ### and status of the jobs will be read from condor log
-                    if jobAd["JobStatus"] == 3:
-                        continue
-                    else:
-                        ## For some strange race condition, schedd sometimes does not publish StartDate for a Running Job
-                        ## Get the entire classad for such a job
-                        ## Do not crash WMA, wait for next polling cycle to get all the info.
-                        if jobAd["JobStatus"] == 2 and jobAd.get("JobStartDate") is None:
-                            logging.debug("THIS SHOULD NOT HAPPEN. JobStartDate is MISSING from the CLASSAD.")
-                            logging.debug("Could be caused by some race condition. Wait for the next Polling Cycle")
-                            logging.debug("%s", str(jobAd))
-                            continue
-
-                        tmpDict = {}
-                        tmpDict["JobStatus"] = int(jobAd.get("JobStatus", 100))
-                        tmpDict["stateTime"] = int(jobAd["EnteredCurrentStatus"])
-                        tmpDict["runningTime"] = int(jobAd.get("JobStartDate", 0))
-                        tmpDict["submitTime"] = int(jobAd["QDate"])
-                        tmpDict["DESIRED_Sites"] = jobAd["DESIRED_Sites"]
-                        tmpDict["ExtDESIRED_Sites"] = jobAd["ExtDESIRED_Sites"]
-                        tmpDict["runningCMSSite"] = jobAd.get("MATCH_EXP_JOBGLIDEIN_CMSSite", None)
-                        tmpDict["WMAgentID"] = int(jobAd["WMAgent_JobID"])
-                        jobInfo[tmpDict["WMAgentID"]] = tmpDict
+            for jobAd in itobj:
+                tmpDict = {}
+                ### This condition ignores jobs that are Removed, but stay in the X state
+                ### For manual condor_rm removal, job wont be in the queue \
+                ### and status of the jobs will be read from condor log
+                if jobAd["JobStatus"] == 3:
+                    continue
+                else:
+                    tmpDict["JobStatus"] = int(jobAd.get("JobStatus", 100))
+                    tmpDict["stateTime"] = int(jobAd["EnteredCurrentStatus"])
+                    tmpDict["runningTime"] = tmpDict["stateTime"] if jobAd['JobStatus'] == 2 else 0
```

Comment:
Actually it's `runningTime`. So we either set it to 0 in case `JobStartDate` is not available or we do some sort of `int(time.time()) - jobAd["EnteredCurrentStatus"]`. Keeping it with 0 for the moment.